src/ML-POD/pair_pod.cpp

```diff
@@ -0,0 +1,370 @@
+
+#include "pair_pod.h"
+
+#include "atom.h"
+#include "comm.h"
+#include "error.h"
+#include "force.h"
+#include "memory.h"
+#include "neigh_list.h"
+#include "neighbor.h"
+#include "tokenizer.h"
+
+#include <glob.h>
+
+#include <cmath>
+#include <cstring>
+#include <sys/time.h>
+
+using namespace LAMMPS_NS;
+
+/* ---------------------------------------------------------------------- */
+
+CPairPOD::CPairPOD(LAMMPS *lmp) : Pair(lmp)
+{
+  single_enable = 0;
+  restartinfo = 0;
+  one_coeff = 1;
+  manybody_flag = 1;
+  centroidstressflag = CENTROID_NOTAVAIL;
+}
+
+/* ---------------------------------------------------------------------- */
+
+CPairPOD::~CPairPOD()
+{
+  free_memory();
+  TemplateFree(gd, backend);
+  TemplateFree(energycoeff, backend);
+  TemplateFree(forcecoeff, backend);
+  TemplateFree(podcoeff, backend);
+  TemplateFree(newpodcoeff, backend);
+
+  delete podptr;
+
+  if (allocated) {
+    memory->destroy(setflag);
+    memory->destroy(cutsq);
+    memory->destroy(scale);
+  }
+}
+
+void CPairPOD::compute(int eflag, int vflag)
+{
+  ev_init(eflag,vflag);
+  vflag_fdotr = 1;
+
+  double **x = atom->x;
+  double **f = atom->f;
+  int **firstneigh = list->firstneigh;
+  int *numneigh = list->numneigh;
+  int *type = atom->type;
+  int *ilist = list->ilist;
+  int nlocal = atom->nlocal;
+  int inum = list->inum;
+  int nall = inum + atom->nghost;
+
+  // initialize global descriptors to zero
+
+  int nd1234 = podptr->pod.nd1234;
+  podptr->podArraySetValue(gd, 0.0, nd1234);
+
+  for (int ii = 0; ii < inum; ii++) {
+    int i = ilist[ii];
+    int jnum = numneigh[i];
+
+    // allocate temporary memory
+
+    if (nijmax < jnum) {
+      nijmax = PODMAX(nijmax, jnum);
```

Comment:
Why use PODMAX and not the MAX macro defined in pointers.h

Reply:
PODMAX is replaced with MAX